### PR TITLE
SSCS-8003 Review confidentiality request

### PR DESCRIPTION
**Pull request by jackmaloney**
https://tools.hmcts.net/jira/browse/SSCS-8003

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 1]
```diff
@@ -163,7 +162,7 @@ protected boolean sendLetterNotification(NotificationWrapper wrapper, Subscripti
     }
 
     private boolean sendMandatoryLetterNotification(NotificationWrapper wrapper, Notification notification, SubscriptionType subscriptionType, Address addressToUse) {
-        if (isMandatoryLetterEventType(wrapper)) {
+        if (MANDATORY_LETTER_EVENT_TYPES.contains(wrapper.getNotificationType())) {
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java:165]: 👍

File: src/main/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterService.java [hunk 2]
```diff
@@ -87,10 +86,8 @@ public PdfLetterService(DocmosisPdfService docmosisPdfService, DocmosisTemplates
         String templatePath = docmosisTemplatesConfig.getCoversheets().get(languagePreference)
                 .get(wrapper.getNotificationType().getId());
         if (StringUtils.isBlank(templatePath)) {
-            throw new PdfGenerationException(
-                    String.format("There is no template for notificationType %s",
-                            wrapper.getNotificationType().getId()),
-                    new RuntimeException("Invalid notification type for docmosis coversheet."));
+            log.info("There is no template for notificationType " + wrapper.getNotificationType().getId());
+            return null;
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterService.java:90]: Is this an error to return null?

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -377,6 +377,11 @@ notification:
         letterId: 5745f77c-6512-4082-8c34-63851f24eab1
       appointee:
         letterId: 114cadaa-7760-4699-9add-d3a252f68909
+    reviewConfidentialityRequest:
```
Comment (ryandac) [inline src/main/resources/application.yaml:380]: should there be an appointee be here?

File: src/test/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapperTest.java [hunk 4]
```diff
@@ -199,44 +196,76 @@ public void givenSubscriptions_shouldGetSubscriptionTypeListWithAppointeeAndJoin
...
                 || type.equals(CASE_UPDATED)
-                || type.equals(RESEND_APPEAL_CREATED_NOTIFICATION)
                 || type.equals(APPEAL_DORMANT_NOTIFICATION)
                 || type.equals(ADJOURNED_NOTIFICATION)
                 || type.equals(APPEAL_RECEIVED_NOTIFICATION)
                 || type.equals(POSTPONEMENT_NOTIFICATION)
                 || type.equals(SUBSCRIPTION_UPDATED_NOTIFICATION)
                 || type.equals(HEARING_BOOKED_NOTIFICATION)
                 || type.equals(STRUCK_OUT)
-                || type.equals(VALID_APPEAL_CREATED)
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapperTest.java:237]: was these removed tests not valid to test?

File: src/test/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterServiceTest.java [hunk 5]
```diff
@@ -113,18 +113,6 @@ public void willCreateAPdfToTheCorrectAddress(final SubscriptionType subscriptio
...
-        NotificationWrapper wrapper = NotificationServiceTest.buildBaseWrapper(
-                APPEAL_LAPSED_NOTIFICATION,
-                appellant,
-                representative,
-                null
-        );
-
-        pdfLetterService.buildCoversheet(wrapper, SubscriptionType.APPELLANT);
-    }
-
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterServiceTest.java:127]: this has answered my previous question

**Review by ryandac — APPROVED** [review]
approved, with a few minor questions

File: src/main/resources/application.yaml (hunk 3 above)
Comment (jackmaloney) [reply src/main/resources/application.yaml:386]: No confidentiality always goes to appellant or joint party

File: src/main/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterService.java (hunk 2 above)
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/docmosis/PdfLetterService.java:90]: This was to fix a problem where it was throwing exception for letters that have no evidence coversheet so should be fine to return null as it's handled elsewhere

**Pull request merged**